src/components/data-processing/DataIntegration/DataIntegration.jsx

```diff
@@ -261,6 +261,21 @@ const DataIntegration = (props) => {
       );
     }
 
+    if ((selectedPlot === 'embedding' || selectedPlot === 'frequency')
+      && !cellSets.loading
+      && cellSets.hierarchy.find((rootNode) => rootNode.key === 'sample')?.children?.length === 1
```

Comment:
Like wise in this one. I don't think it's worth refactoring into a util function on its own though - a small helper function which clarifies the name would be fine.